src/components/TextInput.jsx

```diff
@@ -113,7 +114,7 @@ export default class TextInput extends Component {
     let value = event.target.value;
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
-      value = event.target.value.replace(newLineRegex, ' ');
+      value = value.replace(newLineRegex, ' ');
 
```

Comment:
we can get rid of this state update, check this guy's textInput component: Andrey Popp
his onChange is so simplified.